verilog/analysis/checkers/line_length_rule.cc

```diff
@@ -165,15 +172,75 @@ static bool AllowLongLineException(TokenSequence::const_iterator token_begin,
   return false;
 }
 
+static verible::TokenRange StripNewlineTokens(verible::TokenRange token_range) {
+  // truncate newline tokens and return a new, possibly shrinked range
+  auto reversed = reversed_view(token_range);
+  auto last_non_newline =
+      std::find_if(reversed.begin(), reversed.end(), [](const TokenInfo& t) {
+        return t.token_enum() != TK_NEWLINE;
+      }).base();
+
+  return make_range(token_range.begin(), last_non_newline);
+}
+
+static verible::TokenRange SeekLastNonNewlineToken(
+    verible::TokenRange token_range, TokenSequence::const_iterator text_begin) {
+  // Extend token_range backwards to find non-newline tokens.
+  // text_begin is the frontal limiter that shall not be overrun.
+  // If such a token isn't found, the input range is returned
+  auto found_it = std::find_if(
+      std::reverse_iterator<TokenSequence::const_iterator>(token_range.begin()),
+      std::reverse_iterator<TokenSequence::const_iterator>(text_begin),
+      [](const TokenInfo& t) { return t.token_enum() != TK_NEWLINE; });
+
+  if (found_it !=
+      std::reverse_iterator<TokenSequence::const_iterator>(text_begin)) {
+    // Use the last non-newline token as a new beginning.
+    auto new_begin = found_it.base() - 1;
+    // We move the front backwards, so now it's possible that there are
+    // unneeded tokens in the back.
+    token_range = make_range(new_begin, token_range.end());
+  }
+  return token_range;
+}
+
+verible::TokenRange& CheckTokenRangeForTrailingNewlines(
+    verible::TokenRange* orig_range,
+    TokenSequence::const_iterator text_structure) {
+  // Recall that token_range is *unfiltered* and may contain non-essential
+  // whitespace 'tokens'.
+  // This shrinks the range if there are leading newline tokens
+  *orig_range = StripNewlineTokens(*orig_range);
+  if (orig_range->begin() == orig_range->end()) {
```

Comment:
```suggestion
  if (!std::empty(*orig_range)) return *orig_range; 
```
Maybe something like this? Then just return `StripNewlineTokens(...)` in the other branch.

Reply:
It does not implement enough methods to call either `std::empty` or `TokenRange.empty()` and your proposal (along with variations I have tried) does not compile. The implementation is at `common/util/iterator_range.h`.